packages/playwright-ct-angular/registerSource.mjs

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+// This file is injected into the registry as text, no dependencies are allowed.
+
+/** @typedef {import('../playwright-ct-core/types/component').ObjectComponent} ObjectComponent */
+
+import 'zone.js';
+import { reflectComponentType } from '@angular/core';
+import { getTestBed, TestBed } from '@angular/core/testing';
+import {
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting,
+} from '@angular/platform-browser-dynamic/testing';
+
+/** @type {WeakMap<import('@angular/core/testing').ComponentFixture, Record<string, import('rxjs').Subscription>>} */
+const __pwOutputSubscriptionRegistry = new WeakMap();
+
+/** @type {Map<string, import('@angular/core/testing').ComponentFixture>} */
+const __pwFixtureRegistry = new Map();
+
+getTestBed().initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting(),
+);
+
+/**
+ * @param {ObjectComponent} component
+ */
+async function __pwRenderComponent(component) {
+  const componentMetadata = reflectComponentType(component.type);
+  if (!componentMetadata?.isStandalone)
+    throw new Error('Only standalone components are supported');
+
+  TestBed.configureTestingModule({
+    imports: [component.type],
```

Comment:
If we aim to support components as parameters (for slots?), do we have to list all component types in this `imports` list?

Reply:
Right now, the best option is to let the user create a parent story/test-container component:

```ts
// test-container.ts
@Component({
  imports: [A, B],
  template: `<a/><b/>`
})
class TestContainer {
}

// test
mount(TestContainer);
```

Otherwise, a more convenient option is to allow the option of providing a template to the `mount` function.
```ts
mount(`<a/><b/>`, {imports: [A, B]});
```

Cf. https://github.com/jscutlery/devkit/blob/353aeb322fecad64ba8b3667613deca537adb743/tests/playwright-ct-angular-wide/tests/template.pw.ts#L9

Reply:
This implementation would be more in line with the other frameworks:
https://github.com/sand4rt/playwright-ct-angular/blob/e8a74aa88b88fec329b01661cb0e0235f44d0f5a/ct-angular/tests/slots.spec.ts#L1-L47, but we thought it would be better to merge the library first before moving on to discussing the options